### PR TITLE
Add operations

### PR DESCRIPTION
**Pull request by bmath10**
- This pull request creates a new field in the fhir_endpoints_info and _history tables that includes a list of the endpoint's resource/operation pairs (e.g. [{ "resource": "AllergyInfo", "operation": "read"}, { "resource": "AllergyInfo", "operation": "search-type"},{ "resource": "Device", "operation": "read"}...].
- This also gets rid of the supported_resource field, which wasn't really being used, and all tests have been updated with this new field
- The capability page on the shiny dashboard now allows you to also filter by operation

To migrate the database:
- Do the usual steps (will have to run `make migrate_database` 3 times if you're restoring the database from 2021-02-10 production backup that was posted in the #lantern slack channel)
- Then `make run` and immediately kill the endpoint_manager
- `cd capabilityreceiver/cmd/migrateresources`
- then run `go run main.go up` if you want to do an "up" migration and `go run main.go down` for a "down" migration

----------

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-309
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [ ] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [ ] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [ ] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.

### CODE REVIEW

File: shinydashboard/lantern/fhir_endpoints_fields_json.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ The operation field is an array where an element is an instance of the informati
 * **tls_verison:** Transport Layer Security (TLS) version of the endpoint  
 * **mime_types:** MIME types supported by this endpoint  
 * **updated:** Timestamp of when this endpoint record was updated  
-* **supported_resources:** All of the FHIR resources this endpoint supports  
+* **operation_resource:** An array of pairs of a supported resource and one of it's supported operations.
```
Comment (mayerm94) [inline shinydashboard/lantern/fhir_endpoints_fields_json.md:26]: Can this change be reverted? Looks like the JSON export format hasn't changed.

**Review by mayerm94 — CHANGES_REQUESTED** [review]

File: shinydashboard/lantern/server.R [hunk 2]
```diff
@@ -289,4 +292,54 @@ function(input, output, session) { #nolint
...
+
+  # Resets the display if the user clicks the "Remove All Operations" button
+  observeEvent(input$removeallops, {
+    if (input$removeallops == 0) {
+      return(NULL)
+    }
+    else{
+      updateSelectizeInput(session, "operations",
+              label = "Choose or type in any operation from the list below:",
+              choices = c("all", "read", "vread", "update", "patch", "delete", "history-instance", "history-type", "create", "search-type", "not specified"),
```
[the remark this answers is not in the document]
Comment (bmath10) [reply shinydashboard/lantern/server.R]: good point, removed!

File: shinydashboard/lantern/functions/endpoints.R [hunk 3]
```diff
@@ -141,14 +141,20 @@ get_fhir_resource_types <- function(db_connection) {
...
+get_fhir_resource_by_op <- function(db_connection, field) {
+  res <- tbl(db_connection,
+    sql(paste0("SELECT f.id as endpoint_id,
+      vendor_id,
+      vendors.name as vendor_name,
+      capability_statement->>'fhirVersion' as fhir_version,
+      operation_resource->>'", field, "' as type
+      from fhir_endpoints_info f
+      LEFT JOIN vendors on f.vendor_id = vendors.id
+      WHERE operation_resource != 'null'"))) %>%
```
[the remark this answers is not in the document]
Comment (bmath10) [reply shinydashboard/lantern/functions/endpoints.R]: no you're right, i think this was leftover from the previous version of operation_resource, i've removed it!

File: shinydashboard/lantern/fhir_endpoints_fields_json.md (hunk 1 above)
Comment (bmath10) [reply shinydashboard/lantern/fhir_endpoints_fields_json.md]: yep you're right, i think i got trigger happy with the find and replace, i've switched it back!

**Review by mayerm94 — APPROVED** [review]

**Pull request merged**